src/journal/journalctl.c

```diff
@@ -2661,14 +2692,23 @@ int main(int argc, char *argv[]) {
                         if (n_shown == 0 && !arg_quiet)
                                 printf("-- No entries --\n");
 
-                        if (arg_show_cursor) {
+                        if (arg_show_cursor || arg_cursor_file) {
                                 _cleanup_free_ char *cursor = NULL;
 
                                 r = sd_journal_get_cursor(j, &cursor);
                                 if (r < 0 && r != -EADDRNOTAVAIL)
                                         log_error_errno(r, "Failed to get cursor: %m");
-                                else if (r >= 0)
-                                        printf("-- cursor: %s\n", cursor);
+                                else if (r >= 0) {
+                                        if (arg_cursor_file) {
+                                                r = write_string_file(arg_cursor_file, cursor,
+                                                    WRITE_STRING_FILE_CREATE | WRITE_STRING_FILE_ATOMIC);
+                                                if (r < 0)
+                                                        log_error_errno(r,
+                                                            "Failed to write new cursor to %s: %m",
+                                                            arg_cursor_file);
+                                        } else
+                                                printf("-- cursor: %s\n", cursor);
```

Comment:
this should be shown whenever arg_show_cursor is set. AFAICS in this current version of the patch it wouldn't be if --cursor-file= is also supplied.

Reply:
Yes, you're right. I change this.